integration/volume/volume_test.go

```diff
@@ -0,0 +1,119 @@
+package volume
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	volumetypes "github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/request"
+	"github.com/docker/docker/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVolumesList(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	prefix, _ := getPrefixAndSlashFromDaemonPlatform()
+
+	id := container.Run(t, ctx, client, func(c *container.TestContainerConfig) {
+		c.Config.Volumes = map[string]struct{}{prefix + "foo": {}}
```

Comment:
We could create an helper for that one (because I think it's gonna be use quite a lot)

```go
container.Run(t, ctx, client, WithVolume("source:target"), WithVolume("another:one"))
```

@yongtang @dnephin wdyt ?

Reply:
There are actually two formats. One is `source:target` which is added to `c.HostConfig.Binds`. Another is `path` (without `:`) which is added to `c.Config.Volumes`. Maybe one `WithBind()` and another one `WithVolume()`?

Reply:
`WithBind(src, target string)` `WithVolume(name string)` SGTM

Reply:
Updated. Thanks.